python-package/lightgbm/basic.py

```diff
@@ -512,12 +512,23 @@ def _get_bad_pandas_dtypes(dtypes):
     return bad_indices
 
 
-def _data_from_pandas(data, feature_name, categorical_feature, pandas_categorical):
+def _data_from_pandas(data, feature_name, categorical_feature, pandas_categorical, validate_features=False):
```

Comment:
```suggestion
def _data_from_pandas(data, feature_name, categorical_feature, pandas_categorical, validate_features):
```

In my experience, the use of default values in internal functions can lead to silent bugs of the form "higher-level things forgot to pass through a parameter passed in from the public API".

I think this default should be removed, forcing every call to `_data_from_pandas()` to explicitly choose a value.